tracetools/include/tracetools/tracetools.h

```diff
@@ -404,6 +462,46 @@ _DECLARE_TRACEPOINT(
   const void * rmw_client_handle,
   const char * service_name)
 
+/// `rmw_send_request`
+/**
+ * Request publication.
+ * Notes the pointer to the request being sent at the `rmw` level.
+ * Also notes the sequence number of the request.
+ *
+ * \param[in] rmw_client_handle pointer to the client's `rmw_client_t` handle
+ * \param[in] request pointer to the request being sent
+ * \param[in] sequence_number the sequence number of the request
+ */
+_DECLARE_TRACEPOINT(
+  rmw_send_request,
+  const void * rmw_client_handle,
+  const void * request,
+  int64_t sequence_number)
+
+/// `rmw_take_response`
+/**
+ * Response taking.
+ * Links a `rmw_client_t` handle to a pointer to a response being taken at the `rmw` level.
+ * Notes the source timestamp of the response. Also notes the sequence number of the request this
+ * response is for. It does not note the request's client GID, since it is assumed that the matching
+ * of the response to the original client is performed before this tracepoint.
```

Comment:
same answer as `rmw_send_request`. Due to how the instrumentation works, you're expected to get the mapping from `rmw` handle to client GID from the initialization tracepoint (`rmw_client_init`). Then following the whole communication from the raw trace data is easy enough, although this data is really meant to be processed.

Also, in general, because of the ROS 2 abstraction layers, we can never get just _one_ single "identifier" that you could use/grep to easily see the path of a message/request/response.